runelite-client/src/main/java/net/runelite/client/plugins/ExternalPluginManager.java

```diff
@@ -152,45 +161,182 @@ private void initPluginManager()
+							"%s.gradle.kts",
+							"%s.gradle"
+						);
+
+						@Override
+						public boolean accept(File pathName)
 						{
-							paths.add(file.toPath());
+							// Check if this path looks like a plugin development directory
+							if (!pathName.isDirectory())
+							{
+								return false;
+							}
+
+							String dirName = pathName.getName();
+							if (blacklist.contains(dirName))
+							{
+								return false;
+							}
+
+							boolean isPlugin = false;
+
+							// By convention plugins their directory is $name and they have a $name.gradle.kts or $name.gradle file in their root
+							for (String buildFile : buildFiles)
+							{
+								if (new File(pathName, String.format(buildFile, dirName)).exists())
+								{
+									isPlugin = true;
+									break;
+								}
+							}
+
+							// It is a plugin directory, but we should also check if it can actually be loaded
+							if (!new File(pathName, "build/tmp/jar/MANIFEST.MF").exists())
+							{
+								return false;
+							}
+
+							return isPlugin;
 						}
+					};
 
-						return paths;
+					for (String developmentPluginPath : RuneLiteProperties.getPluginDevelopmentPath())
+					{
+						DevelopmentPluginRepository developmentPluginRepository = new DevelopmentPluginRepository(Paths.get(developmentPluginPath))
+						{
+							@Override
+							public boolean deletePluginPath(Path pluginPath)
+							{
+								// Do nothing, because we'd be deleting our sources!
+								return filter.accept(pluginPath.toFile());
+							}
+						};
+
+						developmentPluginRepository.setFilter(pluginsFilter);
+						compoundPluginRepository.add(developmentPluginRepository);
 					}
-				};
+				}
+
+				return compoundPluginRepository;
 			}
 
 			@Override
 			protected PluginLoader createPluginLoader()
 			{
-				return new JarPluginLoader(this);
+				return new CompoundPluginLoader()
+					.add(new BasePluginLoader(this, new DevelopmentPluginClasspath().addJarsDirectories("build/deps")), this::isDevelopment)
+					.add(new JarPluginLoader(this), this::isNotDevelopment);
 			}
 
 			@Override
-			public RuntimeMode getRuntimeMode()
+			public void loadPlugins()
 			{
-				return RuneLiteProperties.getLauncherVersion() == null ? RuntimeMode.DEVELOPMENT : RuntimeMode.DEPLOYMENT;
+				if (Files.notExists(pluginsRoot) || !Files.isDirectory(pluginsRoot))
+				{
+					log.warn("No '{}' root", pluginsRoot);
+					return;
+				}
+
+				List<Path> pluginPaths = pluginRepository.getPluginPaths();
+
+				if (pluginPaths.isEmpty())
+				{
+					log.warn("No plugins");
+					return;
+				}
+
+				log.debug("Found {} possible plugins: {}", pluginPaths.size(), pluginPaths);
+
+				for (Path pluginPath : pluginPaths)
+				{
+					try
+					{
+						loadPluginFromPath(pluginPath);
+					}
+					catch (PluginRuntimeException e)
+					{
+						if (!(e instanceof PluginAlreadyLoadedException))
+						{
+							log.error("Could not load plugin {}", pluginPath, e);
```

Comment:
When passing an exception to the logger it will always log the message, so `e.getMessage()` doesn't need to be passed to the logger. It was also hard to see which plugin was giving issues, so I added the path.